packages/grafana-data/src/transformations/transformers/reduce.ts

```diff
@@ -224,6 +224,7 @@ export function reduceFields(data: DataFrame[], matcher: FieldMatcher, reducerId
           const value = results[reducer];
           const copy = {
             ...field,
+            type: guessFieldTypeFromValue(value),
```

Comment:
matches the behavior we have in `reduceSeriesToRows(...)` rather than always keeping the same type!